hierarchicalforecast/utils.py

```diff
@@ -642,11 +639,15 @@ def samples_to_quantiles_df(
     )  # [Q,H,N] -> [N,H,Q]
     forecasts_quantiles = forecasts_quantiles.reshape(-1, len(_quantiles))
 
-    df = pd.DataFrame(data=forecasts_quantiles, columns=col_names)
+    df_nw = nw.from_dict(
+        {id_col: unique_ids, time_col: ds, model_name: forecasts_mean},
+        native_namespace=namespace,
+    )
+    df_nw = df_nw.with_columns(**dict(zip(col_names, forecasts_quantiles.T)))
```

Comment:
might be more efficient to create the df in one go
```suggestion
    df_nw = nw.from_dict(
        {
            **{id_col: unique_ids, time_col: ds, model_name: forecasts_mean},
            **dict(zip(col_names, forecasts_quantiles.T)),
        },
        native_namespace=namespace,
    )
```